src/en/juju-upgrade.md

```diff
@@ -0,0 +1,191 @@
+Title: Upgrading Juju software
+
+
+# Upgrading Juju software
+
+At the highest level, a Juju topology can be divided into two main parts:
+
+- Juju client
+- Juju environment (possibly several)
+
+The software at each of the above are managed differently.
+
+The client software is managed by the OS package management system whereas the
+software running in an environment, including the state server (bootstrap
+node), is managed by Juju itself. This section is primarily devoted to
+describing the procedure for upgrading this environment software.
+
+
+## Upgrading the client software
+
+The client software is managed by the OS package management system. With Ubuntu
+this is APT. To upgrade the client, therefore, it is a matter of:
+
+```bash
+sudo apt-get update
+sudo apt-get install juju-core
+```
+
+## Terminology - environment software
+
+Several terms are in circulation that are used to denote the environment software:
+
+- Juju tools
+- Juju agent
+- The daemon/binary, *jujud*
+
+!!! Note: This guide will call "the software running in an environment" the
+**server software**.
+
+
+## Upgrading the server software
+
+Overview:
+
+- The server software is a single binary (`jujud`) that runs on every Juju
+  machine and for every Juju unit. This typically results in multiple binaries
+  per machine. Support for a single binary is in development.
+- The upgrade is initiated via the `upgrade-juju` command.
```

Comment:
fair enough, but maybe a "(discussed below)" would help. Otherwise they may go off looking for it anyhow